intf/intf.go

```diff
@@ -0,0 +1,156 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package intf implements mechanisms to access underlying interfaces in
+// a manner that is agnostic to the underlying implementation. An implementation
+// is included that uses netlink to write to a Linux kernl implementation.
+// Additional implementations can be provided to allow writing to e.g. a
+// gRPCwire implementation.
+package intf
+
+import (
+	"fmt"
+	"net"
+)
+
+// Interface represents information corresponding to a single interface on the
+// underlying system.
+type Interface struct {
+	// Index is an integer index used to refer to the interface.
+	Index int
+	// Name is the name used to refer to the interface in the system.
+	Name string
+	// MAC is the MAC address of the interface.
+	MAC net.HardwareAddr
+}
+
+// ARPEntry is a representation of an ARP entry on the underlying system.
+type ARPEntry struct {
+	// IP is the IP address of the neighbour.
+	IP net.IP
+	// Interface is the name of the interface on which the ARP entry was learnt.
+	Interface string
+	// MAC is the MAC address of the neighbour.
+	MAC net.HardwareAddr
+}
+
+// ARPEvent is used to describe a particular event on the ARP table.
+type ARPEvent int64
+
+const (
+	// ARPUnknown is used when the event is not an add or delete.
+	ARPUnknown ARPEvent = iota
```

Comment:
what does `iota` mean?

Reply:
ah, just looked it up - looks like it increments by 1 for each subsequent const value. 

aside question about Go - how is this a feature in Go? I thought Golang took a minimalist stance, choose to provide only primitives (there isn't even a while loop!). this seems like syntactic sugar that I'd expect from a language like Python